nemo/collections/asr/models/classification_models.py

```diff
@@ -489,95 +647,109 @@ def _update_decoder_config(self, cfg):
 
         OmegaConf.set_struct(cfg, True)
 
-    def _setup_transcribe_dataloader(self, config: Dict) -> 'torch.utils.data.DataLoader':
-        """
-        Setup function for a temporary data loader which wraps the provided audio file.
 
-        Args:
-            config: A python dictionary which contains the following keys:
-            paths2audio_files: (a list) of paths to audio files. The files should be relatively short fragments. \
-                Recommended length per file is between 5 and 25 seconds.
-            batch_size: (int) batch size to use during inference. \
-                Bigger will result in better throughput performance but would use more memory.
-            temp_dir: (str) A temporary directory where the audio manifest is temporarily
-                stored.
+class EncDecRegressionModel(_EncDecBaseModel):
+    """Encoder decoder class for speech regression models.
+    Model class creates training, validation methods for setting up data
+    performing model forward pass.
+    """
 
+    @classmethod
+    def list_available_models(cls) -> List[PretrainedModelInfo]:
+        """
+        This method returns a list of pre-trained model which can be instantiated directly from NVIDIA's NGC cloud.
         Returns:
-            A pytorch DataLoader for the given audio file(s).
+            List of available pre-trained models.
         """
-        dl_config = {
-            'manifest_filepath': os.path.join(config['temp_dir'], 'manifest.json'),
-            'sample_rate': self.preprocessor._sample_rate,
-            'labels': self.cfg.labels,
-            'batch_size': min(config['batch_size'], len(config['paths2audio_files'])),
-            'trim_silence': False,
-            'shuffle': False,
+        result = []
+
+        return result
+
+    def __init__(self, cfg: DictConfig, trainer: Trainer = None):
+        OmegaConf.set_struct(cfg, False)
```

Comment:
I don't think this should be done actually - the user should explicitly set this argument. What can (and should) be done is to check if the `cfg.is_regression_task` value is set to True in the config prior to building the regression model. Else raise a ValueError() stating that the config does not contain the required value of `is_regression_task=True`.

Similarly, the classification model should inspect the config for the absence of `cfg.is_regression_task` or that if it exists in the config, it is False, otherwise raise a ValueError() stating that the config contains an invalid value of `is_regression_task` and it should be False for classification models.

Also, `cfg` is not guarenteed to be `OmegaConf` yet before calling `super().__init__`. Please import model_utils and perform the following

```python
from nemo.utils import  model_utils
...

def __init__(self, cfg: DictConfig, trainer: Trainer = None):
        # Convert config to a DictConfig
        cfg = model_utils.convert_model_config_to_dict_config(cfg)

        # Convert config to support Hydra 1.0+ instantiation
        cfg = model_utils.maybe_update_config_version(cfg)
        
        ....
```